src/seedu/addressbook/Main.java

```diff
@@ -107,7 +107,13 @@ private CommandResult executeCommand(Command command)  {
         try {
             command.setData(addressBook, lastShownList);
             CommandResult result = command.execute();
-            storage.save(addressBook);
+            
+            // Save the addressBook object onto disk and check if the storage file had to be (re)created
+            boolean isFileCreated = storage.save(addressBook);
+            if (isFileCreated) {
```

Comment:
well, this is the right way to do this, but unfortunately, you are not using exception mechanism here; hence defeating the purpose of this exercise :-P